internal/api/hooks.go

```diff
@@ -172,67 +174,80 @@ func (a *API) runHTTPHook(ctx context.Context, r *http.Request, hookConfig conf.
 	return nil, nil
 }
 
-func (a *API) invokeHTTPHook(ctx context.Context, r *http.Request, input, output any) error {
+// invokePostgresHook invokes the hook code. conn can be nil, in which case a new
+// transaction is opened. If calling invokeHook within a transaction, always
+// pass the current transaction, as pool-exhaustion deadlocks are very easy to
+// trigger.
+func (a *API) invokeHook(conn *storage.Connection, r *http.Request, input, output any, uri string) error {
+	var err error
+	var response []byte
+	u, err := url.Parse(uri)
+	if err != nil {
+		return err
+	}
+
 	switch input.(type) {
 	case *hooks.SendSMSInput:
 		hookOutput, ok := output.(*hooks.SendSMSOutput)
 		if !ok {
 			panic("output should be *hooks.SendSMSOutput")
 		}
-		var response []byte
-		var err error
-
-		if response, err = a.runHTTPHook(ctx, r, a.config.Hook.SendSMS, input, output); err != nil {
-			return internalServerError("Error invoking Send SMS hook.").WithInternalError(err)
+		if response, err = a.runHook(r, conn, a.config.Hook.SendSMS, input, output, u.Scheme); err != nil {
+			return err
 		}
-
 		if err := json.Unmarshal(response, hookOutput); err != nil {
 			return internalServerError("Error unmarshaling Send SMS output.").WithInternalError(err)
 		}
+		if hookOutput.IsError() {
```

Comment:
Experimented with moving this check  into the function but am not sure how to maintain the type cast  (e.g. `hooks.SendSMSOutput`) while keeping the inputs for `runHook` generic. 

Tried using  `HookOutput` as the type for the input parameter but could not keep `runHook` generic while covering all  extension points